### PR TITLE
Adding JFrog Artifactory

### PR DESCRIPTION
**Pull request by jainishshah17**

### CODE REVIEW

**Comment by viglesiasce** [comment]
Hey @jainishshah17!!

Thanks for the submission!

A few things to fix up:
1. The nginx chart was errantly checked in
2. You need to sign the Linux Foundation CLA as noted above
3. The .gitignore change has already been checked in
4. Use a ClusterIP service by default, we prefer not to optimize for Minikube
5. Does this chart work out of the box without having a pro license?
6. You have a note about volumes and config maps in the deployment but I dont see that configuration included.

File: incubator/artifactory/README.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+Deletion of the PetSet doesn't cascade to deleting associated Pods and PVCs. To delete them:
+
+```
+$ kubectl delete pods -l release=my-release,type=data
+$ kubectl delete pvcs -l release=my-release,type=data
+```
+
+## Configuration
+
+The following tables lists the configurable parameters of the elasticsearch chart and their default values.
```
Comment (ultimateboy) [inline incubator/artifactory/README.md]: s/elasticsearch/JFrog Artifactory

Comment (jainishshah17) [reply incubator/artifactory/README.md]: Fixed

**Comment by chrislovecnm** [comment]
Can u squash your commits amigo?

**Comment by chrislovecnm** [comment]
@jainishshah17 your rebase is a tad wonky. Please remove the .ignore changes out of the commit ;)

File: incubator/artifactory/README.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
+# JFrog Artifactory Helm Chart
+
+## Prerequisites Details
+
+* Kubernetes 1.3 with alpha APIs enabled
+* Artifactory Pro trial license
```
Comment (chrislovecnm) [inline incubator/artifactory/README.md]: @viglesiasce how are we handling paid for software? Do we need some notice in the README, hey you need to pay for this ;)

File: incubator/artifactory/README.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+$ helm install --name my-release incubator/artifactory
+```
+
+## Deleting the Charts
+
+Deletion of the PetSet doesn't cascade to deleting associated Pods and PVCs. To delete them:
+
+```
+$ kubectl delete pods -l release=my-release,type=data
+$ kubectl delete pvcs -l release=my-release,type=data
```
Comment (chrislovecnm) [inline incubator/artifactory/README.md]: Can you include the timed wait that is in the PetSet docs?

File: incubator/artifactory/templates/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        # Making image configurable is not necessary. Making imageTag configurable
+        # is a nice option for the user. Especially in the strange cases like
+        # artifactory where the base distro is determined by the tag. Using :latest
+        # is frowned upon, using :stable isn't that great either.
+        image: "{{default "docker.bintray.io/jfrog/artifactory-pro" .Values.image}}:{{default "4.13.2" .Values.imageTag}}"
+        imagePullPolicy: {{default "IfNotPresent" .Values.pullPolicy}}
+        ports:
+        - containerPort: 8081
+          name: http
+        # This (and the volumes section below) mount the config map as a volume.
```
Comment (chrislovecnm) [inline incubator/artifactory/templates/deployment.yaml]: Where is this at? We need resource limit and health probes as well please

**Review by chrislovecnm — CHANGES_REQUESTED** [review]
Thanks for the PR!! Have a couple tweaks

File: incubator/artifactory/README.md (hunk 3 above)
Comment (jainishshah17) [reply incubator/artifactory/README.md]: @chrislovecnm  I have updated my chart

**Comment by prydonius** [comment]
@jainishshah17 any luck on the Linux Foundation CLA?

**Comment by jainishshah17** [comment]
@prydonius Done signing it.

**Comment by prydonius** [comment]
Can you rebase to fix the conflicts?

File: incubator/artifactory/README.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+## Configuration
+
+The following tables lists the configurable parameters of the artifactory chart and their default values.
+
+|         Parameter         |           Description             |                         Default                          |
+|---------------------------|-----------------------------------|----------------------------------------------------------|
+| `Image`                   | Container image name              | `docker.bintray.io/jfrog/artifactory-pro`                |
+| `ImageTag`                | Container image tag               | `4.13.2`                                                 |
+| `ImagePullPolicy`         | Container pull policy             | `Always`                                                 |
```
Comment (gtaylor) [inline incubator/artifactory/README.md]: Sorry to come in late, but can this be left blank to fall through to the Kubernetes default of IfNotExists, like here? https://github.com/kubernetes/charts/blob/master/stable/redis/templates/deployment.yaml#L19

File: .gitignore [hunk 6]
```diff
@@ -1,35 +0,0 @@
-# General files for the project
```
Comment (chrislovecnm) [inline .gitignore:1]: You need to remove this file from your commit.  You are removing it from the repo.  Add it back in your branch please.

File: incubator/artifactory/README.md (hunk 3 above)
Comment (chrislovecnm) [reply incubator/artifactory/README.md]: No here is an example

``` console
$ grace=$(kubectl get po web-0 --template '{{.spec.terminationGracePeriodSeconds}}')
$ kubectl delete petset,po -l app=nginx
$ sleep $grace
$ kubectl delete pvc -l app=nginx
```

File: incubator/artifactory/templates/deployment.yaml [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+        app: {{template "fullname" .}}
+        release: {{.Release.Name | quote }}
+    spec:
+      containers:
+      - name: artifactory
+        # Making image configurable is not necessary. Making imageTag configurable
+        # is a nice option for the user. Especially in the strange cases like
+        # artifactory where the base distro is determined by the tag. Using :latest
+        # is frowned upon, using :stable isn't that great either.
+        image: "{{default "docker.bintray.io/jfrog/artifactory-pro" .Values.image}}:{{default "4.13.2" .Values.imageTag}}"
```
Comment (chrislovecnm) [inline incubator/artifactory/templates/deployment.yaml]: I would prefer to have these hard coded values in values.yaml.  @viglesiasce thoughts?

File: incubator/artifactory/templates/deployment.yaml [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+        imagePullPolicy: {{default "IfNotPresent" .Values.pullPolicy}}
+        resources:
```
Comment (chrislovecnm) [inline incubator/artifactory/templates/deployment.yaml:22]: This is a much cleaner way to do the resources https://github.com/kubernetes/charts/blob/master/stable/mysql/templates/deployment.yaml#L34

You will need the https://github.com/kubernetes/charts/blob/master/stable/mysql/templates/_helpers.tpl file as well in your chart.

File: incubator/artifactory/templates/deployment.yaml [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+      volumes:
+        - name: etc
+        configMap:
+          name: {{ template "fullname" . }}
+        - name: data
+        configMap:
+          name: {{ template "fullname" . }}
+        - name: logs
+        configMap:
+          name: {{ template "fullname" . }}
```
Comment (chrislovecnm) [inline incubator/artifactory/templates/deployment.yaml]: You list PVC in your instructions, but you do not define any PVC components in your chat?

**Review by chrislovecnm — CHANGES_REQUESTED** [review]
Thanks for the chart.  Some more notes.

File: incubator/artifactory/values.yaml [hunk 10]
```diff
@@ -0,0 +1,17 @@
+# Default values for etcd.
```
Comment (alon-argus) [inline incubator/artifactory/values.yaml]: etcd?

File: incubator/artifactory/templates/deployment.yaml (hunk 7 above)
Comment (alon-argus) [reply incubator/artifactory/templates/deployment.yaml]: I agree. Also, fix to `.Values.Image` (capital i) and `.Values.ImageTag`, as in `values.yaml`.

File: incubator/artifactory/templates/deployment.yaml [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+        release: {{.Release.Name | quote }}
+    spec:
+      containers:
+      - name: artifactory
+        # Making image configurable is not necessary. Making imageTag configurable
+        # is a nice option for the user. Especially in the strange cases like
+        # artifactory where the base distro is determined by the tag. Using :latest
+        # is frowned upon, using :stable isn't that great either.
+        image: "{{default "docker.bintray.io/jfrog/artifactory-pro" .Values.image}}:{{default "4.13.2" .Values.imageTag}}"
+        imagePullPolicy: {{default "IfNotPresent" .Values.pullPolicy}}
```
Comment (alon-argus) [inline incubator/artifactory/templates/deployment.yaml]: `.Values.pullPolicy` -> `.Values.ImagePullPolicy`

File: incubator/artifactory/templates/svc.yaml [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+kind: Service
+metadata:
+  name: {{template "fullname" .}}
+  labels:
+    heritage: {{ .Release.Service | quote }}
+    release: {{ .Release.Name | quote }}
+    chart: "{{.Chart.Name}}-{{.Chart.Version}}"
+spec:
+  ports:
+  - port: {{default 8081 .Values.httpPort}}
```
Comment (alon-argus) [inline incubator/artifactory/templates/svc.yaml:12]: `.Values.httpPort` is not defined

**Comment by viglesiasce** [comment]
Can you add a [NOTES.txt](https://github.com/kubernetes/helm/blob/master/docs/charts.md#chart-license-readme-and-notes) file that explains what users need to do in order to use the Chart?

**Comment by jainishshah17** [comment]
@prydonius I have updated my changes. Please check

**Comment by viglesiasce** [comment]
@k8s-bot ok to test

File: stable/mysql/values.yaml [hunk 13]
```diff
@@ -31,7 +31,7 @@ imageTag: "5.7.14"
 persistence:
   enabled: true
   storageClass: generic
-  accessMode: ReadWriteOnce
+  accessMode: ReadWrite
```
Comment (viglesiasce) [inline stable/mysql/values.yaml]: This should not be changed in this PR.

File: incubator/artifactory/values.yaml [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+resources:
+  requests:
+    memory: 2048Mi
+    cpu: 200m
+
+## Persist data to a persitent volume
+persistence:
+  enabled: true
+  storageClass: generic
+  accessMode: ReadWrite
```
Comment (viglesiasce) [inline incubator/artifactory/values.yaml]: Invalid access mode, see test results.

File: incubator/artifactory/values.yaml [hunk 15]
```diff
@@ -0,0 +1,27 @@
+# Default values for Artifactory.
+# This is a YAML-formatted file.
+# Declare name/value pairs to be passed into your templates.
+# name: value
+
+Name: artifactory
+Component: "Artifactory"
+image: "docker.bintray.io/jfrog/artifactory-pro"
```
Comment (viglesiasce) [inline incubator/artifactory/values.yaml]: can we default this to the oss image and have a note about how to switch it?

**Review by viglesiasce — CHANGES_REQUESTED** [review]

File: stable/mysql/values.yaml (hunk 13 above)
Comment (jainishshah17) [reply stable/mysql/values.yaml]: Changed default image to Artifactory-oss

**Comment by jainishshah17** [comment]
@k8s-bot e2e test this

**Comment by viglesiasce** [comment]
@k8s-bot ok to test

File: incubator/artifactory/templates/NOTES.txt [hunk 16]
```diff
@@ -0,0 +1,6 @@
+Artifactory can be accessed via port 8081 on the following DNS name from within your cluster:
+{{ template "fullname" . }}.{{ .Release.Namespace }}.svc.cluster.local
+
```
Comment (viglesiasce) [inline incubator/artifactory/templates/NOTES.txt:19]: Add examples of how to connect to this from outside the cluster. Check out the jenkins chart for an example of how to differentiate between the various service modes.

File: incubator/artifactory/values.yaml [hunk 17]
```diff
@@ -0,0 +1,29 @@
...
+## Uncomment following line if you want to run Artifactory-Pro
+#image: "docker.bintray.io/jfrog/artifactory-pro"
+imageTag: "4.14.1"
+imagePullPolicy: "Always"
+replicaCount: 1
+httpPort: 8081
+## Kubernetes configuration
+## For minikube, set this to NodePort, elsewhere use LoadBalancer
+##
+ServiceType: LoadBalancer
```
Comment (viglesiasce) [inline incubator/artifactory/values.yaml]: Lets make this ClusterIP by default since the password isnt randomly generated by default.

**Review by viglesiasce — CHANGES_REQUESTED** [review]

**Comment by viglesiasce** [comment]
Looking good @jainishshah17! Was able to install it this time around. 2 more things to fix:

1. Add NOTES about how to connect from outside the cluster
2. Make the service ClusterIP by default

**Comment by jainishshah17** [comment]
@viglesiasce Made changes please check

**Review by viglesiasce — APPROVED** [review]

**Comment by viglesiasce** [comment]
This worked for me. Thanks @jainishshah17!

**Pull request merged**